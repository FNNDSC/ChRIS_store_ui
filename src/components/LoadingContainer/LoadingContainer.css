/* loading class if non-white, used for loading content */
.loading-container.loading, .loading-container .loading {
  /* color */
  background: #f5f5f5;
  background: var(--pf-black-150);
  
  /* position */
  position: relative;
  
  /* other */
  box-shadow: 1px 0 2px #ededed;
  box-shadow: 1px 0 2px var(--pf-black-200);
  overflow: hidden;
}

.loading-container.loading:before,
.loading-container .loading:before {
  /* color */
  background-image: linear-gradient(45deg, #f5f5f5, #ededed, #f5f5f5);
  background-image: linear-gradient(45deg, var(--pf-black-150), var(--pf-black-200), var(--pf-black-150));

  /* display */
  display: block;
  height: 100%;
  width: 100%;

  /* position */
  position: absolute;
  top: 0;
  left: 0;

  /* other */
  content: '';   
  animation: 1.6s ease-in-out loading infinite;
}

/* loading class if white, used for loading content */
.loading-container.loading, .loading-container .loading.white {
  /* color */
  background: #fafafa;
  background: var(--pf-black-100);
  
  /* other */
  box-shadow: 1px 0 2px #f5f5f5;
  box-shadow: 1px 0 2px var(--pf-black-150);
  overflow: hidden;
}

.loading-container.loading.white:before,
.loading-container .loading.white:before {
  /* color */
  background-image: linear-gradient(45deg, #fafafa, #f0f0f0, #fafafa);
  background-image: linear-gradient(45deg, var(--pf-black-100), #f0f0f0, var(--pf-black-100));

  /* display */
  display: block;
  height: 100%;
  width: 100%;

  /* position */
  position: absolute;
  top: 0;
  left: 0;

  /* other */
  content: '';   
  animation: 1.6s ease-in-out loading infinite;
}

/* loading keyframe */
@keyframes loading {
  0% { left: -100%; }
  10% { left: -100%; }
  90% { left: 100%; }
  100% { left: 100%; }
}