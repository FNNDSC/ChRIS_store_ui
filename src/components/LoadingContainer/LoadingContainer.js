import React from 'react';
import PropTypes from 'prop-types';
import './LoadingContainer.css';

const LoadingContainer = props => (
  <div className={`loading-container ${props.className}`}>
    {props.children}
  </div>
);

LoadingContainer.propTypes = {
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

LoadingContainer.defaultProps = {
  className: '',
  children: null,
};

export default LoadingContainer;
