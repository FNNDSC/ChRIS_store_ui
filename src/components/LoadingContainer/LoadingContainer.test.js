import React from 'react';
import { shallow } from 'enzyme';
import LoadingContainer from './LoadingContainer';

describe('LoadingContainer', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<LoadingContainer />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render loading-container div', () => {
    expect(wrapper.find('div.loading-container')).toHaveLength(1);
  });

  it('should render className prop', () => {
    wrapper.setProps({ className: 'testClass' });
    expect(wrapper.find('div.loading-container.testClass')).toHaveLength(1);
  });
});

const withChildren = (
  <LoadingContainer>
    <div className="test-div" />
  </LoadingContainer>
);

describe('rendered LoadingContainer', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(withChildren);
  });

  it('should render children prop', () => {
    expect(wrapper
      .find('div.loading-container')
      .find('div.test-div'))
      .toHaveLength(1);
  });
});
