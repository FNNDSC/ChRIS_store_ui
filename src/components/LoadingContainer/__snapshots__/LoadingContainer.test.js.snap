// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingContainer should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoadingContainer
    className=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": null,
      "className": "loading-container ",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": null,
        "className": "loading-container ",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
